code/controllers/subsystem/materials.dm

```diff
@@ -7,27 +7,27 @@ These materials call on_applied() on whatever item they are applied to, common e
 
 SUBSYSTEM_DEF(materials)
 	name = "Materials"
-	flags = SS_NO_FIRE
-	init_order = INIT_ORDER_MATERIALS
+	flags = SS_NO_FIRE | SS_NO_INIT
 	///Dictionary of material.type || material ref
-	var/list/materials = list() 
+	var/list/materials = list()
 	///Dictionary of category || list of material refs
-	var/list/materials_by_category = list() 
+	var/list/materials_by_category = list()
 	///List of stackcrafting recipes for materials using rigid materials
 	var/list/rigid_stack_recipes = list(
 		new /datum/stack_recipe("chair", /obj/structure/chair/greyscale, one_per_turf = TRUE, on_floor = TRUE, applies_mats = TRUE),
 		new /datum/stack_recipe("toilet", /obj/structure/toilet/greyscale, one_per_turf = TRUE, on_floor = TRUE, applies_mats = TRUE),
 		new /datum/stack_recipe("sink", /obj/structure/sink/greyscale, one_per_turf = TRUE, on_floor = TRUE, applies_mats = TRUE),
 	)
 
-/datum/controller/subsystem/materials/Initialize(timeofday)
-	InitializeMaterials()
-	return ..()
-
 ///Ran on initialize, populated the materials and materials_by_category dictionaries with their appropiate vars (See these variables for more info)
-/datum/controller/subsystem/materials/proc/InitializeMaterials(timeofday)
+/datum/controller/subsystem/materials/proc/InitializeMaterials()
 	for(var/type in subtypesof(/datum/material))
 		var/datum/material/ref = new type
 		materials[type] = ref
 		for(var/c in ref.categories)
 			materials_by_category[c] += list(ref)
+
+/datum/controller/subsystem/materials/proc/GetMaterialRef(datum/material/fakemat)
+	if(!materials)
+		InitializeMaterials()
+	return materials[fakemat] || fakemat
```

Comment:
this is TG. I'm pretty sure TG loves like defines and shizz.

```byond
#define GET_MATERIAL_REF(fakemat) ((SSmaterials.materials ? InitializeMaterials() : null) || (SSmaterials.materials[fakemat] || fakemat))
```

this is only half a shitpost.

Reply:
I considered it but there are usecases we would need a proc for that we've been discussing for the future